frontend/src/views/student/tournament/EditTournamentView.vue

```diff
@@ -0,0 +1,358 @@
+      sortable: false
+    }
+  ];
+
+  async created() {
+    this.editTournament = this.tournament;
+    if (this.editTournament.startTime) {
+      this.oldStartTime = this.newStartTime = this.editTournament.startTime;
+    }
+    if (this.editTournament.endTime) {
+      this.oldEndTime = this.newEndTime = this.editTournament.endTime;
+    }
+    if (this.editTournament.numberOfQuestions) {
+      this.oldNumberOfQuestions = this.editTournament.numberOfQuestions;
+    }
+    this.oldTopics = this.editTournament.topics!;
+    await this.$store.dispatch('loading');
+    try {
+      [this.allTopics] = await Promise.all([
+        RemoteServices.getTournamentTopics()
+      ]);
+      this.availableTopics = this.allTopics;
+      if (this.editTournament.topics !== undefined) {
+        this.editTournament.topics!.forEach(topicName => {
+          this.availableTopics!.forEach(topic => {
+            if (topic.name.valueOf() == topicName.valueOf()) {
+              this.addTopic(topic);
+            }
+          });
+        });
+      }
+    } catch (error) {
+      await this.$store.dispatch('error', error);
+    }
+    await this.$store.dispatch('clearLoading');
+  }
+
+  async resetChanges() {
+    this.editTournament.startTime = this.oldStartTime;
+    this.editTournament.endTime = this.oldEndTime;
+    this.editTournament.numberOfQuestions = this.oldNumberOfQuestions;
+    this.editTournament.topics = this.oldTopics;
+  }
+
+  async cancelTournament() {
+    await this.resetChanges();
+    this.$emit('close-edit-dialog');
+  }
+
+  async saveTournament() {
+    this.editTournament.startTime = this.newStartTime;
+    this.editTournament.endTime = this.newEndTime;
+
+    if (
+      this.editTournament &&
+      (!this.editTournament.startTime ||
+        !this.editTournament.endTime ||
+        !this.editTournament.numberOfQuestions ||
+        this.currentTopics.length == 0)
+    ) {
+      await this.$store.dispatch(
+        'error',
+        'Tournament must have Start Time, End Time, Number Of Questions and Topics'
+      );
+      await this.resetChanges();
+      return;
+    }
+
+    if (this.editTournament && this.editTournament.id != null) {
+      const participants = this.editTournament.participants;
+      this.editTournament.participants = [];
+
+      let topicsList = this.currentTopics.map(topic => {
+        return topic.id;
+      });
+
+      try {
+        const result = await RemoteServices.updateTournament(
+          topicsList,
+          this.editTournament
+        );
+        this.$emit('edit-tournament', result);
+      } catch (error) {
+        await this.$store.dispatch('error', error);
+        this.editTournament.participants = participants;
+      }
+      this.editTournament.topics = this.currentTopics.map(topic => {
+        return topic.name;
+      });
+      this.editTournament.participants = participants;
+    }
+  }
+
+  topicFilter(value: string, search: string, topic: Topic) {
```

Comment:
there is a lot of repeated code with CreateTournamentView... can we have a single view that is reused in both cases?

Reply:
Existem diferenças significativas para que justifique serem 2 views diferentes

Reply:
Which are? what about do the remote access outside and reuse the manipulation of the tournament?